superset/sql/parse.py

```diff
@@ -1421,10 +1406,12 @@ def extract_tables_from_jinja_sql(sql: str, database: Database) -> set[Table]:
             node.data = "NULL"
 
     # re-render template back into a string
-    rendered_template = Template(template).render(processor.get_context())
+    code = processor.env.compile(ast)
+    template = Template.from_code(processor.env, code, globals=processor.env.globals)
+    rendered_sql = template.render(processor.get_context())
```

Comment:
Made sure to include the fix from https://github.com/apache/superset/pull/33596.